playwright/fabric-custom.test.ts

```diff
@@ -0,0 +1,56 @@
+import { expect, test } from '@playwright/test';
+
+const viewport = { width: 1600, height: 1000 };
+
+const widths = ['360', '740', '980', '1300', '100%'];
+
+test.describe('Fabric Custom visual regression testing', () => {
+	test('Get reference screenshots', async ({ page }) => {
+		await page.setViewportSize(viewport);
+
+		await page.goto(
+			'https://guardian.github.io/commercial-templates/ssr/fabric-custom/',
+			{
+				waitUntil: 'networkidle',
+			},
+		);
+
+		for (const breakpoint of widths) {
+			const referenceTemplateLocator = page
+				.frameLocator(`[name='width-${breakpoint}']`)
+				.locator('#creative');
+			// check that the template is present on the page
+			await referenceTemplateLocator.isVisible();
```

Comment:
aiui `isVisible` will evaluate and return immediately rather than waiting. To reduce flakiness its preferable to use `expect(locator).toBeVisible()`

https://playwright.dev/docs/api/class-locator#locator-is-visible

An example:

https://github.com/guardian/dotcom-rendering/blob/db040dce572a80e6d33c71be1b1df006604a6024/dotcom-rendering/playwright/lib/locators.ts#L6

Reply:
Oh I didn't realise that! Good catch, thanks Ravi